WordPress/src/main/java/org/wordpress/android/ui/mysite/MySiteCardAndItem.kt

```diff
@@ -389,8 +391,39 @@ sealed class MySiteCardAndItem(open val type: Type, open val activeQuickStartIte
             val onCtaClick: ListItemInteraction,
         ) : Card(type = Type.WP_SOTW_2023_NUDGE_CARD)
 
-        data class NoCardsMessage(val title: UiString, val message: UiString)  : Card(Type.NO_CARDS_MESSAGE)
+        data class NoCardsMessage(val title: UiString, val message: UiString) : Card(Type.NO_CARDS_MESSAGE)
         data class PersonalizeCardModel(val onClick: () -> Unit) : Card(Type.PERSONALIZE_CARD)
+
+        data class Dynamic(
+            val id: String,
+            val rows: List<Row>,
+            val title: String?,
+            val image: String?,
+            val action: ActionSource?,
+            val onHideMenuItemClick: ListItemInteraction,
+        ) : Card(type = DYNAMIC_DASHBOARD_CARD) {
+            data class Row(
+                val iconUrl: String?,
+                val title: String?,
+                val description: String?,
+            )
+
+            sealed class ActionSource {
+                abstract val url: String
+                abstract val onCtaClick: ListItemInteraction
+
+                data class Card(
+                    override val url: String,
+                    override val onCtaClick: ListItemInteraction
+                ) : ActionSource()
+
+                data class Button(
+                    override val url: String,
+                    override val onCtaClick: ListItemInteraction,
+                    val title: String
+                ) : ActionSource()
+            }
```

Comment:
According to p1702484824130069/1702473203.210019-slack-C067H36F011:
- URL is mandatory for a CTA
- CTA button is visible when there is a title
- Card CTA is available for actions with no title
- onCtaClick property available for non-null action only

Reply:
Thank you for implementing this Artyom 🙇